lib/email/receiver.rb

```diff
@@ -745,23 +745,23 @@ def process_destination(destination, user, body, elided)
 
     def create_group_post(group, user, body, elided)
       message_ids = Email::Receiver.extract_reply_message_ids(@mail, max_message_id_count: 5)
-      post_ids = []
 
+      # incoming emails with matching message ids, and then cross references
+      # these with any email addresses for the user vs to/from/cc of the
+      # incoming emails. in effect, any incoming email record for these
+      # message ids where the user is involved in any way will be returned
       incoming_emails = IncomingEmail.where(message_id: message_ids)
       if !group.allow_unknown_sender_topic_replies
         incoming_emails = incoming_emails.addressed_to_user(user)
       end
-
-      incoming_emails = incoming_emails.pluck(:post_id, :from_address, :to_addresses, :cc_addresses)
-
-      incoming_emails.each do |post_id, from_address, to_addresses, cc_addresses|
```

Comment:
I removed this completely because, unless I am gravely mistaken, it is doing the exact same thing as this:

https://github.com/discourse/discourse/blob/1bfde69180226e344dba2359ecfe47046d8eda68/app/models/incoming_email.rb#L21-L32